src/changelog.ts

```diff
@@ -242,20 +267,41 @@ export async function generateChangelog(
         continue;
       }
 
-      if (!category["messages"]) {
-        category["messages"] = [];
+      if (changelog.get(scope)?.get(category.title) === undefined) {
+        changelog.get(scope)?.set(category.title, []);
       }
-      category["messages"].push(await generateChangelogEntry(commit.message));
+
+      changelog
+        .get(scope)
+        ?.get(category.title)
+        ?.push(await generateChangelogEntry(commit.message));
+
       break;
     }
   }
 
+  // Sort changelog, with the all (*) scope always as last item
+  changelog = new Map(
+    [...changelog].sort((a, b) =>
+      a[0] === "*" ? 1 : b[0] === "*" ? -1 : String(a[0]).localeCompare(b[0])
+    )
+  );
```

Comment:
A map is an unordered collection, so using it to store a sorted list into is tricky. You can keep the array with the map spread and use it as-is.

```suggestion
const sortedChangelog = [...changelog].sort((a, b) =>
      a[0] === "*" ? 1 : b[0] === "*" ? -1 : a[0].localeCompare(b[0])
  );
```
(Also, `a[0]` is already a `string` there) :)